### PR TITLE
Reduce logging in statestore - redis

### LINKED ISSUE #1228: Unnecessary logging when waiting for assignments is canceled.
Currently, when a waiting for assignment, a client will start the call and receive a message when the assignment has been updated.  Then it drops the call, because it doesn't need it anymore.

On the next iteration of the wait for assignment loop, it then logs in two places (at least) that the context is canceled:
https://github.com/googleforgames/open-match/blob/master/internal/statestore/redis.go#L584

https://github.com/googleforgames/open-match/blob/master/internal/statestore/redis.go#L228

This is partially a result of over logging that needs to be cleaned up as part of https://github.com/googleforgames/open-match/issues/1102

However, there's another issue here:
We've now stated that getting an assignment is a terminal state for a ticket (besides being deleted), so there's no point in the wait for assignment call to continue querying the database.  It will always get the same answer until it errors due to the ticket not being there, or canceled by the caller.

Should the FE change so that it will terminate the call once it successfully gets an assignment?  I think that behavioral change falls into the backwards compatibility promises for 1.0, but just barely.

Otherwise, perhaps we need to both: Fix the over logging that's occurring, and suppress logging for canceled context on wait for assignment, though that starts to get iffy on suppressing errors the user might care about.  We could get smart about when to suppress (eg, only after assignment was found), but that's getting even more magical and weird.

### PR DESCRIPTION
**Pull request by akremsa**
Closes #1228 

Previously _context canceled_ message was mentioned twice: 
1) connect method
2) GetAssignments method, after GetTicket returns an error.
Logs before the fix:
image

I suggest to log everything on a very top level of API and just wrap everything which comes from lower levels.
Also, I've made logs more informative adding an ID to each record.

Logs after the fix:
image

### CODE REVIEW

File: internal/statestore/redis.go [hunk 1]
```diff
@@ -223,18 +223,22 @@ func redisURLFromAddr(addr string, cfg config.View, usePassword bool) string {
...
 
 // CreateTicket creates a new Ticket in the state storage. If the id already exists, it will be overwritten.
 func (rb *redisBackend) CreateTicket(ctx context.Context, ticket *pb.Ticket) error {
+	if ticket == nil {
+		return status.Error(codes.InvalidArgument, "nil ticket provided")
+	}
+
 	redisConn, err := rb.connect(ctx)
 	if err != nil {
+		redisLogger.WithFields(logrus.Fields{
```
Comment (Laremere) [inline internal/statestore/redis.go:239]: Per the linked issue: https://github.com/googleforgames/open-match/issues/1102

It's not good to both log an error and also return it, this is the cause of the initial issue.

File: internal/statestore/redis.go [hunk 2]
```diff
@@ -335,8 +345,15 @@ func (rb *redisBackend) DeleteTicket(ctx context.Context, id string) error {
 
 // IndexTicket indexes the Ticket id for the configured index fields.
 func (rb *redisBackend) IndexTicket(ctx context.Context, ticket *pb.Ticket) error {
+	if ticket == nil {
```
Comment (Laremere) [inline internal/statestore/redis.go:348]: We already have validation in the calling code, eg https://github.com/googleforgames/open-match/blob/master/internal/app/frontend/frontend_service.go

Here and elsewhere in your changes we don't need to be validating at this level.  If there's any missing validation, it should be added in the grpc service.

**Comment by akremsa** [comment]
@Laremere PR is updated, please check.
The current output is:
image

**Comment by Laremere** [comment]
Great work!  Please follow up with cleaning up elsewhere (the non-streaming calls).

**Pull request merged**